connectors/utils.py

```diff
@@ -372,6 +372,25 @@ class ConcurrentTasks:
 
     - `max_concurrency`: max concurrent tasks allowed, default: 5
     - `results_callback`: when provided, synchronous function called with the result of each task.
+
+    Examples:
+
+        # create a task pool with the default max concurrency
+        task_pool = ConcurrentTasks()
+
+        # put a task into pool
+        # it will block until the task was put successfully
+        task = task_pool.put(a_coroutine)
+
+        # put a task without blocking
+        # it will try to put the task, and return None if it can't be put immediately
+        task = task_pool.try_put(a_coroutine)
+
+        # call join to wait for all tasks in pool to complete
+        # this is not required to execute the tasks in pool
+        # a task will be automatically scheduled to execute once it's put successfully
+        # call join() only when you need to do something after all tasks in pool complete
```

Comment:
What kind of weird results do you expect? 

Calling `join` will just wait for all tasks (at the moment) to be completed before returning.

Assuming there are 5 tasks in the pool, and you call `join`, it will wait for the 5 tasks to complete. Before it returns, if you add the 6th task, the 6th task should remain in the pool.

Let me see if I can add a test for this case.

Reply:
Yeah exactly this case - adding 5 tasks, joining, then adding another one and expecting all 6 to finish. We had this bug and it was actually a part of implementation/tests at some point, as far as I remember.

Reply:
Hi @artem-shelkovnikov , added the test case here: https://github.com/elastic/connectors/pull/1479/commits/e2e0197e6974004b9603d15456c8638e1f43dae0